wallet-new/test/Integration/Specs/Addresses.hs

```diff
@@ -0,0 +1,25 @@
+module Integration.Specs.Addresses (spec) where
+
+import           Universum
+
+import           Cardano.Wallet.Client (WalletClient)
+import           Cardano.Wallet.Client.Http
+import           Test.Hspec (Spec, it)
+
+import           Integration.Expectations
+import           Integration.Fixtures
+import           Integration.Util (runSpec)
+
+
+spec :: WalletClient IO -> Spec
+spec apiClient =
+    it "Creating an address makes it available" $ runSpec apiClient $ do
+        (account, wallet) <- getArbitraryAccount
+
+        let body = NewAddress
+                { newaddrSpendingPassword = Nothing
+                , newaddrAccountIndex     = accIndex account
+                , newaddrWalletId         = walId wallet
+                }
+
+        lift (postAddress apiClient body) >>= addressShouldExists
```

Comment:
Now the actual spec, written using the availables DSL and split in basically three parts:

1 - setup / fixture
2 - action
3 - assertions

In theory, we would have a last _tear-down_ which is in reality a bit complex with a blockchain (immutable data). It would require us to reset each node between each step. This takes a couple of seconds each time, so we may want to think about this. In theory, tests shouldn't depend on each other and could be ran in any order. We can probably preserve this without resetting the state between each step by being careful about the inputs we take (especially the amount of coins when we make transactions).

Reply:
I feel like we could avoid the `lift` by having a `MonadIO m => WalletClient m` instead of an `IO` client directly.

Reply:
Sounds like a fair point :D